db/users/update.go

```diff
@@ -47,6 +47,17 @@ func UpdateUser(ctx context.Context, app, addonUUID, username string) error {
 	if !userExists {
 		return errors.New(ctx, fmt.Sprintf("User \"%s\" does not exist", username))
 	}
+	// Check if the user is protected
```

Comment:
issue(blocking): you can avoid a loop here by saving the user in the last for loop (https://github.com/Scalingo/cli/blob/2b8d41467bf986a62428527f4232f6cd5fa4c36e/db/users/update.go#L41).
And then do you check after that.